lib/apiservers/engine/backends/image.go

```diff
@@ -67,44 +64,18 @@ func (i *Image) ImageHistory(imageName string) ([]*types.ImageHistory, error) {
 }
 
 func (i *Image) Images(filterArgs string, filter string, all bool) ([]*types.Image, error) {
+	layerCache.Lock()
+	defer layerCache.Unlock()
 
 	result := []*types.Image{}
 
-	host, err := guest.UUID()
-	if host == "" {
-		host, err = os.Hostname()
-	}
-	if err != nil {
-		return result,
-			derr.NewErrorWithStatusCode(fmt.Errorf("image.Images got unexpected error getting hostname"),
-				http.StatusInternalServerError)
-	}
-
-	params := storage.NewListImagesParams().WithStoreName(host)
-
-	client := PortLayerClient()
-	if client == nil {
-		return result,
-			derr.NewErrorWithStatusCode(fmt.Errorf("image.Images failed to create a portlayer client"),
-				http.StatusInternalServerError)
-	}
-
-	layers, err := client.Storage.ListImages(params)
-	if err != nil {
-		log.Warning(err)
-		return result, nil
-	}
-
-	// TODO(jzt): avoid this by having portlayer send back only images (not layers)
-	// find actual image layers
-	for _, layer := range layers.Payload {
+	for _, layer := range layerCache.cache {
 		imageConfig := &metadata.ImageConfig{}
 		if err := json.Unmarshal([]byte(layer.Metadata["metaData"]), imageConfig); err != nil {
```

Comment:
Why not have an image cache?  Then the marshaling could be done during cache update and not on read...